framework/src/modules/dpos_v2/method.ts

```diff
@@ -58,4 +64,29 @@ export class DPoSMethod extends BaseMethod {
 
 		return delegate;
 	}
+
+	public getRoundLength(_methodContext: ImmutableMethodContext): number {
+		return this._config.roundLength;
+	}
+
+	public getNumberOfActiveDelegates(_methodContext: ImmutableMethodContext): number {
+		return this._config.numberActiveDelegates;
+	}
+
+	public async updateSharedRewards(
+		_methodContext: MethodContext,
+		_generatorAddress: Buffer,
+		_tokenID: Buffer,
+		_reward: bigint,
+	): Promise<void> {
+		// TODO: Implement #7715
+	}
+
+	public isEndOfRound(_methodContext: ImmutableMethodContext, height: number): boolean {
```

Comment:
I think that `isLastBlockOfRound()` would be a more accurate name. The function can take any block height, from present, past or future. All it does is calculate if the given height belongs to the last block of the round.

To make my point even stronger, I saw that in `module.ts` you even assigned the result of this function to a variable named `isLastBlockOfRound` 🙂

Also, there is a much cleaner way to calculate the last block of the round, without using `Rounds` and creating all these new variables. It needs 1 line of code only:

```typescript
return height % this._config.roundLength === 0
```

That is if the first block has height 1. Otherwise, if the first block has height 0:

```typescript
return (height + 1) % this._config.roundLength === 0
```

Reply:
While continuing the code review, I noticed that it wasn't you who originally wrote this function. You just moved it from `DPoSModule` to `DPoSMethod`... and decided to rename it for some reason 🙂

Reply:
The name was defined in the [LIP](https://research.lisk.com/t/introduce-dynamic-block-rewards-module/387#after-transactions-execution-42) as `method`, so I used as it is.
The method itself has been used for a while, so i didn't change it.
round needs to consider genesis height, so the calculation should be 
```
(h - genesisDataStore.height) % ROUND_LENGTH == 0
```
Ill update the logic